able to update username, email and password

fnoori opened the pull request.

Commit:
able to update username, email and password

Commit:
fixed lint errors

nxtman123 commented:
In your comments describing the arguments for the the event/function, make sure to mention all required or possible arguments including `confirm_email`, etc.

Before checking all the details about one of the fields, make sure to test if the new value is different from the old value. If it is the same, silently skip it. This will prevent a possible error when it is found that a user (this user, in fact) already has that email/username.

Also, check that the username and email are not too long for their db fields.

Commit:
Implemented recomendations

Needed to add underscorejs to check for JSON emptiness
  tried to do this the js way, but lint was throwing errors

nxtman123 commented on src/server/authentication.js:
```diff
@@ -87,6 +92,118 @@ module.exports = psql => ({
+    try {
+      const toUpdate = {};
+      const userInDatabase = await psql('users')
+        .where('user_id', '=', user.user_id);
```
Append .first() to just get the one row outside of an array (and avoid [0] everywhere)

nxtman123 commented on src/server/authentication.js:
```diff
@@ -87,6 +92,118 @@ module.exports = psql => ({
+        .where('user_id', '=', user.user_id);
+
+      if (updateData.email) {
+        if (updateData.email.length > EMAIL_MAX) {
```
This test could be after-inside the test for if this is a real change

fnoori replied:
If I were to put it inside the test to see if it is a real change, I would need to potentially query the db three times, once for each check: email, username, and password.

The way I have it right now, it just needs to query the db once (drawback being, it queries even if the changes are not real).

nxtman123 replied:
I had not thought of that. Good call

nxtman123 reviewed (commented):
Just a couple nit-picks 😛 merge when you're ready

Commit:
Using knex first() function to stop using [0] to access the first element
